Modules/Scripted/SegmentStatistics/SegmentStatisticsCalculators/SegmentStatisticsCalculatorBase.py

```diff
@@ -0,0 +1,201 @@
+import vtk
+import qt
+import slicer
+
+
+class SegmentStatisticsCalculatorBase(object):
+  """Base class for statistics calculators operating on segments.
+  Derived classes should specify: self.name, self.keys, self.defaultKeys
+  and implement: computeStatistics, getMeasurementInfo
+  """
+
+  @staticmethod
+  def initCodedEntry(codeValue, codingScheme, codeMeaning, returnAsString=False):
+    entry = slicer.vtkCodedEntry()
+    entry.SetValueSchemeMeaning(codeValue, codingScheme, codeMeaning)
+    return entry if not returnAsString else entry.GetAsString()
+
+  @staticmethod
+  def getDICOMTriplet(codeValue, codingSchemeDesignator, codeMeaning):
+    """Utility method to form DICOM triplets in getMeasurementInfo"""
+    return {'CodeValue':codeValue,
+            'CodingSchemeDesignator': codingSchemeDesignator,
+            'CodeMeaning': codeMeaning}
+
+  @staticmethod
+  def generateMeasurementInfo(name, description, units, quantityCode, unitsCode, measurementMethodCode=None,
+                              derivationCode=None):
+    info = {
+      "name": name,
+      "description": description,
+      "units": units,
+      "DICOM.QuantityCode": quantityCode,
+      "DICOM.UnitsCode": unitsCode
+      }
+    if measurementMethodCode:
+      info["DICOM.MeasurementMethodCode"] = measurementMethodCode
+    if derivationCode:
+      info["DICOM.DerivationCode"] = derivationCode
+    return info
+
+  def __init__(self):
+    self.name = "" # name of the statistics calculator
```

Comment:
Documentation of base plugin class may be used by developers, so it would be useful if you could document it in a way to facilitate automatic documentation generation later.
Sphinx expects member variable documentation in the line above the assignment, prefixed by ```# :``` (see http://www.sphinx-doc.org/en/stable/ext/autodoc.html#directive-autoattribute)